ext/panko_serializer/time_conversion.c

```diff
@@ -1,65 +1,79 @@
 #include "time_conversion.h"
 
+const int YEAR_REGION = 1;
+const int MONTH_REGION = 2;
+const int DAY_REGION = 3;
+const int HOUR_REGION = 4;
+const int MINUTE_REGION = 5;
+const int SECOND_REGION = 6;
+
 static regex_t* iso8601_time_regex;
 static regex_t* ar_iso_datetime_regex;
 
 VALUE is_iso8601_time_string(const char* value) {
   const UChar *start, *range, *end;
+  OnigPosition r;
 
   const UChar* str = (const UChar*)(value);
 
   end = str + strlen(value);
   start = str;
   range = end;
-  OnigPosition r = onig_search(iso8601_time_regex, str, end, start, range, NULL,
-                               ONIG_OPTION_NONE);
+  r = onig_search(iso8601_time_regex, str, end, start, range, NULL,
+                  ONIG_OPTION_NONE);
 
   return r >= 0 ? Qtrue : Qfalse;
 }
 
-void append_region_part(char* buf,
-                        const char* value,
-                        const OnigRegion* region,
-                        int i) {
-  long substringLength = region->end[i] - region->beg[i];
-  strncat(buf, value + region->beg[i], substringLength);
+void append_region(const char* source,
+                   char** to,
+                   const OnigRegion* region,
+                   int region_number) {
+  long iter = 0;
+  for (iter = region->beg[region_number]; iter < region->end[region_number];
```

Comment:
Just a suggestion. `region->end[region_number]` will be called each iteration. If there are more than 2 or 3 iterations it might make sense to make a local variable for the value.

Reply:
I will make it a local value, just because the line is long.
We do here 4 iterations at max.